### ISSUE TITLE
show dart packages dependencies in the explorer view

### ISSUE THREAD

**Issue by devoncarew** [Dart-Code/Dart-Code#348]
I think this could be done by parsing the .packages file, listening for changes, and contributing a new explorer view and TreeDataProvider:

```
		"views": {
			"explorer": [
				{
					"id": "dartPackages",
					"name": "Dart Packages",
					"when": "resourceLangId == 'dart'"
				}
			]
		},
```

**Comment by DanTup** [comment]
I've merged this into a branch and changed the context to always when a Dart project is loaded. I agree that opening the packages as editable sucks, so I've raised a request code Code to support readonly files: https://github.com/Microsoft/vscode/issues/31410

**Comment by devoncarew** [comment]
👍 cool!

**Comment by DanTup** [comment]
Looks like I misunderstood, and the text content providers aren't required to be HTML (the sample code I saw was relying on Code being clever and guessing content type based on the first line, which was an HTML tag).

Should be trivial to switch this over, and then maybe it'll be reasonable to ship.

**Comment by DanTup** [comment]
@devoncarew Made some tweaks to this (inc. readonly) and shipped in v2.3.2. Thanks!
